### ISSUE TITLE
Advanced Search

### ISSUE THREAD

**Issue by markus4000** [osTicket/osTicket-1.8#1662]
How can I search for all open tickets with 1.9.5 ?
problem: if I have open tickets with different statuses: 
Like "open", "approved", "on hold", "preview" etc.. then I cannot filter all the open tickets. I can only search for "open" or for approved etc, but not for all really open tickets in the same search query.
is there a way?

**Comment by greezybacon** [comment]
Not in v1.9.5. In the next version of osTicket, probably v1.9.6, the advanced search system is rewritten and this will be possible by searching based on the ticket state (not status) of "open"

**Comment by humangenome** [comment]
Any filter I place in the Advanced Search window yields the same result: all tickets (Open, Closed, all statuses) - seems it does not work at all right now. Using latest Github release from today.

**Comment by greezybacon** [comment]
> Any filter I place in the Advanced Search window yields the same result: all tickets (Open, Closed, all statuses) - seems it does not work at all right now. Using latest Github release from today.

That isn't true. The advanced search does not work when searching for "Assigned Agent". Other criteria work correctly.

**Comment by markus4000** [comment]
In 196 it still does not work. 
Is there a quickfix so i can search in all "state open" ?
I tried by my self by adding a checkbox to the advanved search, but i couldnt find where to get the post variable and where to implement in tp the sql query.

**Comment by ntozier** [comment]
This works in 1.9.5.1 for me.
1.9.6 hasn't really been released yet (not on web page, and no announcement).

**Comment by markus4000** [comment]
Ntozier. I think you have only one status with the state open in your installation. Or all the state open tickets ar in status open.

State is not equal to status.
I need a way to search in all statuses that are of the state open. Else, if there are more than one status in state open, i cannot find out how many tickets team xyz has to work on it, with only one search request. I have to do a search for every status.

**Comment by markus4000** [comment]
solved for 1.9.6 with a little mod:
in File: ajax.tickets.php
Line159 (insection "Flags")
i added: 

                case 'open':
                    $criteria['state'] = 'open';
                    $where .= ' AND status.state="open" ';
                  break;

and in File ticket.inc.php 
around line 602 in the advanced form filed set "Flags"
i added the Line:

<option data-state="open" value="open"><?php echo __('Open Tickets');?></option>

Now I can search in the "Open" state, by using the flag filter "open tickets"

**Comment by markus4000** [comment]
In 1.9.7 it ist still not possible (native) but with that simple quickfix here, its possible to search only in the open AND only the closed tickets too. Very helpful if you have to search for something or to set a filter for statistics. etc. specialy if you work with several diffrent ticketstatuses

in file: include/ajax.tickets.php
Line around 159 (in section "Flags")
add: 

                case 'open':
                    $criteria['state'] = 'open';
                    $where .= ' AND status.state="open" ';
                    break;
                case 'closed':
                    $criteria['state'] = 'closed';
                    $where .= ' AND status.state="closed" ';
                break; 


 and in file include/staff/ticket.inc.php around line 602 in the advanced form filed set "Flags"
 add the Lines : 

                <option data-state="open" value="open"><?php echo __('Open Tickets');?></option>  
                <option data-state="close" value="closed"><?php echo __('Closed Tickets');?></option>

**Comment by greezybacon** [comment]
@markus4000 I agree, such an enhancement would be nice for the current version of osTicket.

**Comment by fictionedge** [comment]
I had a bit of trouble finding the second part, for those who had the same problem (I'm using v.1.9.9) this is how is looks afterwords.

```
<fieldset class="span6">
            <label for="flag"><?php echo __('Flags');?>:</label>
            <select id="flag" name="flag">
                 <option value="">&mdash; <?php echo __('Any Flags');?> &mdash;$
                 <?php
                 if (!$cfg->showAnsweredTickets()) { ?>
                 <option data-state="open" value="answered"><?php echo __('Answ$
                 <?php
                 } ?>
                 <option data-state="open" value="overdue"><?php echo __('Overd$
                 <option data-state="open" value="open"><?php echo __('Open Tic$

```

**Comment by markus4000** [comment]
This is how it Looks in my code (its from1.9.8) :


        <fieldset class="span6">
            <label for="flag"><?php echo __('Flags');?>:</label>
            <select id="flag" name="flag">
                 <option value="">&mdash; <?php echo __('Any Flags');?> &mdash;</option>
                 <?php
                 if (!$cfg->showAnsweredTickets()) { ?>
                 <option data-state="open" value="answered"><?php echo __('Answered');?></option>
                 <?php
                 } ?>
                 <option data-state="open" value="overdue"><?php echo __('Overdue');?></option>
                 <option data-state="open" value="open"><?php echo __('Open Tickets');?></option>
                 <option data-state="close" value="closed"><?php echo __('Closed Tickets');?></option>
            </select>
        </fieldset>
